README.md

```diff
@@ -145,6 +152,117 @@ readData(dataInput) {
 }
 ```
 
+# Using Node.js Client with Hazelcast IMDG
+
+## 1. Node.js Client API Overview
+
+The first step is configuration. You can configure the Node.js client declaratively or programmatically. We will use the programmatic approach throughout this chapter. Please refer to the [Node.js Client Declarative Configuration section](#declarative-configuration) for details.
+
+```javascript
+var clientConfig = new Config.ClientConfig();
+clientConfig.groupConfig.name = 'dev';
+clientConfig.networkConfig.addresses.push('10.90.0.1', '10.90.0.2:5702');
+```
+
+The second step is to initialize the `HazelcastClient` to be connected to the cluster.
+
+```javascript
+Client.newHazelcastClient(clientConfig).then(function (client) {
+    // some operation
+});
+```
+
+**This client object is your gateway to access all Hazelcast distributed objects.**
+
+Let’s create a map and populate it with some data.
+
+```javascript
+var mapCustomers;
+Client.newHazelcastClient(clientConfig).then(function (client) {
+    mapCustomers = client.getMap('customers'); // creates the map proxy
+    return mapCustomers.put('1', new Customer('Furkan', 'Senharputlu'));
+}).then(function () {
+    return mapCustomers.put('2', new Customer("Joe", "Smith"));
+}).then(function () {
+    return mapCustomers.put('3', new Customer("Muhammet", "Ali"));
+});
+```
+
+As a final step, if you are done with your client, you can shut it down as shown below. This will release all the used resources and will close connections to the cluster.
+
+```javascript
+...
+.then(function () {
```

Comment:
When I see this snippet, I would assume that I can just add this as a last step to the above chain, it would work. However, it will not work because `client` is defined in scope of only the first closure above. `client` here would be undefined and error

Reply:
I will add a `var client;` in the first snippet.